addons/mail/static/src/discuss/core/common/composer_patch.js

```diff
@@ -0,0 +1,10 @@
+/* @odoo-module */
+
+import { Composer } from "@mail/core/common/composer";
+import { patch } from "@web/core/utils/patch";
+
+patch(Composer.prototype, {
+    get allowUpload() {
+        return super.allowUpload && (this.thread?.allow_public_upload || this.store.user);
```

Comment:
Is store.user filled or not for portal? I'm not too sure.

Reply:
It is. I updated the condition to take that into account.